remi/server.py

```diff
@@ -172,7 +173,10 @@ def bytetonum(self,b):
 
     def read_next_message(self):
         debug_message('read_next_message\n')
-        length = self.rfile.read(2)
+        try:
+            length = self.rfile.read(2)
+        except timeout:
```

Comment:
I considered it, but then I need to do:

    import socket

Up on the file, which may import many things. I wanted to skip that. If you consider that is better to do it that way I'll do.

But please, before even considering it, try it with some code you have for a bit of time! This is quite a critical point in the code and there may be a reason this was this way (other than technological debt).